### PR TITLE
parser: support plan replayer dump explain file

### PR DESCRIPTION
**Pull request by Yisaer**
Signed-off-by: yisaer <disxiaofei@163.com>

### What problem does this PR solve?

Issue Number: ref #37798 

Problem Summary:

### What is changed and how it works?

Support plan replayer dump explain file

### Check List

Tests 

- [x] Unit test

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: parser/parser.y [hunk 1]
```diff
@@ -14241,6 +14242,26 @@ PlanReplayerStmt:
...
+		}
+		$$ = x
+	}
+|	"PLAN" "REPLAYER" "DUMP" "EXPLAIN" "ANALYZE" stringLit
+	{
+		x := &ast.PlanReplayerStmt{
+			Stmt:    nil,
+			Analyze: true,
+			Load:    false,
+			File:    $6,
```
Comment (xuyifangreeneyes) [inline parser/parser.y:14263]: In `PLAN REPLAYER LOAD 'dump_file'`, we set `PlanReplayerStmt.File` to `dump_file`. Maybe we can add comment to explain different usages of `PlanReplayerStmt.File` between `PLAN REPLAYER DUMP` and `PLAN REPLAYER LOAD` or we can use different fields for different usages.

**Review by xuyifangreeneyes — COMMENTED** [review]
I have one question that may be not very related to the pr: if there are some sqls which cannot be explained, do we return error or just skip them? Currently it seems to return syntax error. Rest LGTM.

**Comment by Yisaer** [comment]
> I have one question that may be not very related to the pr: if there are some sqls which cannot be explained, do we return error or just skip them? Currently it seems to return syntax error. Rest LGTM.

Currently we will return error now.

**Review by xuyifangreeneyes — APPROVED** [review]

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**